code/GridFieldDetailFormPreviewExtension.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace SilverStripe\Admin;
+
+use SilverStripe\Admin\LeftAndMain;
+use SilverStripe\CMS\Controllers\SilverStripeNavigator;
+use SilverStripe\Core\Extension;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\LiteralField;
+use SilverStripe\ORM\CMSPreviewable;
+use SilverStripe\View\SSViewer;
+
+class GridFieldDetailFormPreviewExtension extends Extension
+{
+    public function updateItemEditForm(Form $form)
+    {
+        $record = $this->owner->getRecord();
+        if ($record instanceof CMSPreviewable) {
+            $form->addExtraClass('cms-previewable');
+            $navField = LiteralField::create('SilverStripeNavigator', $this->getSilverStripeNavigator());
+            $navField->setAllowHTML(true);
+            $form->Fields()->push($navField);
+        }
+    }
```

Comment:
This is basically a direct copy of https://github.com/silverstripe/silverstripe-admin/blob/d3fc48cde0aef36518d0b7efd11dd5711bef6be4/code/LeftAndMain.php#L1393-L1398, just for context.